doc/source/starting-ray.rst

```diff
@@ -0,0 +1,105 @@
+Starting Ray
+============
+
+This page covers how to start Ray on your single machine or cluster of machines.
+
+.. contents:: :local:
+
+Installation
+------------
+
+Install Ray with ``pip install -U ray``. For the latest wheels (for a snapshot of ``master``), you can use these instructions at :ref:`install-nightlies`.
+
+Starting Ray on a single machine
+--------------------------------
+
+You can start Ray by calling ``ray.init()`` in your Python script. This needs to happen before any Ray tasks or actors are called (i.e., ``function.remote()`` will not work).
+
+.. code-block:: python
+
+  import ray
+  ray.init()
+
+To stop or restart Ray, use ``ray.shutdown()``. This is useful when Ray is initialized improperly:
+
+.. code-block:: python
+
+  import ray
+  ray.init()
+  ... # ray program
+  ray.shutdown()
+
+
+To check if Ray is initialized, you can call ``ray.is_initialized()``:
+
+.. code-block:: python
+
+  import ray
+  ray.init()
+  assert ray.is_initialized() == True
+
+  ray.shutdown()
+  assert ray.is_initialized() == False
+
+See the `Configuration <configure.html>`__ documentation for the various ways to configure Ray.
+
+Using Ray on a cluster
+----------------------
+
+There are two steps needed to use Ray in a distributed setting:
+
+    1. You must first start the Ray cluster.
+    2. You need to add the ``address`` parameter to ``ray.init`` (like ``ray.init(address=...)``).
+
+If you have a Ray cluster specification (:ref:`ref-automatic-cluster`), you can launch a multi-node cluster with Ray initialized on each node with ``ray up``. **From your local machine/laptop**:
+
+.. code-block:: bash
+
+    ray up cluster.yaml
+
+You can monitor the Ray cluster status with ``ray monitor cluster.yaml`` and ssh into the head node with ``ray attach cluster.yaml``.
+
+Your Python script will **only** need to execute on the same machine as the head node. To make sure your Ray program utilizes all cluster resources, add the following to your Python script:
+
+.. code-block:: python
+
+    ray.init(address="auto")
+
+.. note:: Without ``ray.init(address...)``, your Ray program will only be parallelized across a single machine!
+
+Manual Cluster Setup
+~~~~~~~~~~~~~~~~~~~~
+
+You can also use the manual cluster setup (:ref:`ref-cluster-setup`) by running initialization commands on each node.
+
+**On the head node**:
+
+.. code-block:: bash
+
+    # If the ``--redis-port`` argument is omitted, Ray will choose a port at random.
+    $ ray start --head --redis-port=6379
+
+The command will print out the address of the Redis server that was started (and some other address information).
+
+**Then on all of the other nodes**, run the following. Make sure to replace ``<address>`` with the value printed by the command on the head node (it should look something like ``123.45.67.89:6379``).
+
+.. code-block:: bash
+
+    $ ray start --address=<address>
+
+
+Turning off parallelism
```

Comment:
Not sure that we want to advertise this so prominently given that we put very little effort into supporting it (we should probably have a separate discussion about this).

Reply:
added a caution statement